### PR TITLE
fix: Orthogonal seeding beamspot and function

### PR DESCRIPTION
**Pull request by CarloVarni**
The orthogonal seed finder was creating internal space points imposing a beam spot position of `{0, 0}` which is hardly the case if we are not running on MC. 

Also, adding a function for extracting global position and variance from the external space point, instead on assuming methods. The function has to be user-provided.

### CODE REVIEW

File: Examples/Scripts/Python/full_chain_itk.py [hunk 1]
```diff
@@ -12,6 +12,7 @@
 )
 from acts.examples.reconstruction import (
     addSeeding,
+    SeedingAlgorithm,
```
Comment (stephenswat) [inline Examples/Scripts/Python/full_chain_itk.py:15]: Is this a related change that needs to be included here?

**Review by stephenswat — CHANGES_REQUESTED** [review]
Excellent idea, must absolutely be implemented. I would like to see the function objects passed as arbitrary template parameters.

File: Core/include/Acts/Seeding/SeedFinderOrthogonal.hpp [hunk 2]
```diff
@@ -100,10 +101,14 @@ class SeedFinderOrthogonal {
...
+   * @param extract_coordinates User-defined function for extracting global position and
+   * covariance of the external space point
    */
   template <typename input_container_t, typename output_container_t>
-  void createSeeds(const input_container_t &spacePoints,
-                   output_container_t &out_cont) const;
+  void createSeeds(
+      const input_container_t &spacePoints, output_container_t &out_cont,
+      std::function<std::pair<Acts::Vector3, Acts::Vector2>(
+          typename input_container_t::value_type)> &extract_coordinates) const;
```
[the remark this answers is not in the document]
Comment (CarloVarni) [reply Core/include/Acts/Seeding/SeedFinderOrthogonal.hpp]: Thanks for the comment. I should have implemented your suggestion. Take a look at the new diff to confirm you are are satisfied with the new changes.

File: Examples/Scripts/Python/full_chain_itk.py (hunk 1 above)
Comment (CarloVarni) [reply Examples/Scripts/Python/full_chain_itk.py:15]: This is for selecting the seeding algorithm (e.g. Default vs Orthogonal) in the job option. It is not really connected to the PR title though. I'd keep it here without opening a different PR for this additional trivial change

**Review by stephenswat — APPROVED** [review]
Perfect, thank you!
